contracts/test/integration/IntegrationTestVip13.t.sol

```diff
@@ -0,0 +1,220 @@
+        vm.stopPrank();
+
+        uint256 balance = usdc.balanceOf(MainnetAddresses.KRAKEN_USDC_WHALE);
+        vm.prank(MainnetAddresses.KRAKEN_USDC_WHALE);
+        usdc.transfer(address(this), balance);
+
+        vm.startPrank(MainnetAddresses.GOVERNOR);
+        core.grantMinter(MainnetAddresses.GOVERNOR);
+        voltV2.mint(address(voltV2UsdcPriceBoundPSM), voltMintAmount);
+        oldVolt.mint(address(this), voltMintAmount);
+        voltV2.mint(address(this), voltMintAmount);
+        vm.stopPrank();
+
+        usdc.transfer(address(voltV2UsdcPriceBoundPSM), balance / 2);
+    }
+
+    function testSwapDaiForVolt() public {
+        uint256 currentPegPrice = oracle.getCurrentOraclePrice();
+        uint256 userStartingVoltBalance = voltV2.balanceOf(address(this));
+        uint256 minAmountOut = voltV2DaiPriceBoundPSM.getMintAmountOut(
+            mintAmountDai
+        );
+        uint256 startingPSMUnderlyingBalance = dai.balanceOf(
+            address(voltV2DaiPriceBoundPSM)
+        );
+
+        dai.approve(address(voltV2DaiPriceBoundPSM), mintAmountDai);
+        uint256 amountVoltOut = voltV2DaiPriceBoundPSM.mint(
+            address(this),
+            voltMintAmount,
+            minAmountOut
+        );
+
+        uint256 endingUserVOLTBalance = voltV2.balanceOf(address(this));
+        uint256 endingPSMUnderlyingBalance = dai.balanceOf(
+            address(voltV2DaiPriceBoundPSM)
+        );
+
+        assertEq(
+            endingUserVOLTBalance,
+            amountVoltOut + userStartingVoltBalance
+        );
+        assertEq(
+            endingPSMUnderlyingBalance - startingPSMUnderlyingBalance,
+            mintAmountDai
+        );
+        assertApproxEq(
+            ((mintAmountDai * 1e18) / currentPegPrice).toInt256(),
+            minAmountOut.toInt256(),
+            0
+        );
+    }
+
+    function testSwapVoltForDai() public {
+        uint256 currentPegPrice = oracle.getCurrentOraclePrice();
+        uint256 startingUserUnderlyingBalance = dai.balanceOf(address(this));
+        uint256 startingPSMUnderlyingBalance = dai.balanceOf(
+            address(voltV2DaiPriceBoundPSM)
+        );
+
+        uint256 redeemAmountOut = voltV2DaiPriceBoundPSM.getRedeemAmountOut(
+            voltMintAmount
+        );
+        uint256 startingUserVOLTBalance = voltV2.balanceOf(address(this));
+
+        voltV2.approve(address(voltV2DaiPriceBoundPSM), voltMintAmount);
+        uint256 amountOut = voltV2DaiPriceBoundPSM.redeem(
+            address(this),
+            voltMintAmount,
+            redeemAmountOut
+        );
+
+        uint256 endingUserVOLTBalance = voltV2.balanceOf(address(this));
+        uint256 endingUserUnderlyingBalance = dai.balanceOf(address(this));
+        uint256 endingPSMUnderlyingBalance = dai.balanceOf(
+            address(voltV2DaiPriceBoundPSM)
+        );
+
+        assertEq(
+            startingUserVOLTBalance,
+            endingUserVOLTBalance + mintAmountDai
```

Comment:
unsure why this assertion is in place. Shouldn't it be `startingVoltBalance - voltMintAmount == endingVoltBalance`?

Reply:
or

`startingUserVOLTBalance == endingUserVOLTBalance + voltMintAmount`

like in the next test that does a redemption against usdc